### PR TITLE
OAS-2264 | Support for accepting the T&C for deployment resource & set default OASIS_ENDPOINT to production environment

### PR DESCRIPTION
**Pull request by Robert-Stam**

### CODE REVIEW

File: pkg/operational_test_resources.go [hunk 1]
```diff
@@ -34,25 +36,11 @@ import (
...
-	} else if len(organizations.GetItems()) < 1 {
-		client.log.Error().Err(err).Msg("No organizations found")
-		return "", fmt.Errorf("no organizations found")
-	} else {
-		return organizations.GetItems()[0].GetId(), nil
+	if _, ok := os.LookupEnv("OASIS_TEST_ORGANIZATION_ID"); !ok {
+		return "", fmt.Errorf("This test requires an organization id to be set.")
 	}
+	orgID := os.Getenv("OASIS_TEST_ORGANIZATION_ID")
+	return orgID, nil
```
Comment (ewoutp) [inline pkg/operational_test_resources.go:43]: Both calls do the same.
Can be simplified to:

```
orgID := os.Getenv("OASIS_TEST_ORGANIZATION_ID")
if orgID == "" return error
```

File: pkg/resource_deployment_test.go [hunk 2]
```diff
@@ -18,19 +18,54 @@
...
+func TestResourceDeployment(t *testing.T) {
+	if _, ok := os.LookupEnv("TF_ACC"); !ok {
+		t.Skip()
+	}
+	t.Parallel()
+
+	res := "terraform-deployment-" + acctest.RandString(10)
+	name := "deployment-" + acctest.RandString(10)
+	orgID, err := FetchOrganizationID(testAccProvider)
+	assert.NoError(t, err)
```
Comment (ewoutp) [inline pkg/resource_deployment_test.go:50]: -> require.NoError

File: pkg/resource_deployment_test.go [hunk 3]
```diff
@@ -18,19 +18,54 @@
...
+	pid, err := FetchProjectID(orgID, testAccProvider)
+	assert.NoError(t, err)
```
Comment (ewoutp) [inline pkg/resource_deployment_test.go:52]: -> require.NoError

File: pkg/resource_deployment_test.go [hunk 4]
```diff
@@ -189,3 +224,43 @@ func TestExpandDeploymentOverrideProjectID(t *testing.T) {
...
+	return fmt.Sprintf(`resource "oasis_deployment" "%s" {
+	terms_and_conditions_accepted = "true"
+	name        = "%s"
+	description = "Terraform Generated Deployment"
+	project     = "%s"
+	location {
+	  region = "gcp-europe-west4"
+	}
+	version {
+	  db_version = "3.6.0"
```
Comment (ewoutp) [inline pkg/resource_deployment_test.go:238]: That seems too old.
How can we keep this current?
Perhaps envvar?

**Review by ewoutp — COMMENTED** [review]
Few issues

File: pkg/operational_test_resources.go (hunk 1 above)
Comment (Robert-Stam) [reply pkg/operational_test_resources.go:43]: fixed

File: pkg/resource_deployment_test.go (hunk 2 above)
Comment (Robert-Stam) [reply pkg/resource_deployment_test.go:50]: fixed

File: pkg/resource_deployment_test.go (hunk 3 above)
Comment (Robert-Stam) [reply pkg/resource_deployment_test.go:52]: fixed (2)

File: pkg/resource_deployment_test.go (hunk 4 above)
Comment (Robert-Stam) [reply pkg/resource_deployment_test.go:239]: 3.6.0 is available for _support, so OK for internal testing.

Comment (Robert-Stam) [reply pkg/resource_deployment_test.go:239]: I wondered why the version is required, if we make it optional, we can simple use the default version?

**Review by ewoutp — APPROVED** [review]
LGTM

File: pkg/resource_deployment_test.go (hunk 4 above)
Comment (Robert-Stam) [reply pkg/resource_deployment_test.go]: as discussed made version optional (and defaults to the current)

**Pull request merged**